client/src/partials/vouchers/toolkit/convention_payment.js

```diff
@@ -0,0 +1,190 @@
+angular.module('bhima.controllers')
+.controller('ConventionPaymentKitController', ConventionPaymentKitController);
+
+// DI definition
+ConventionPaymentKitController.$inject = [
+  '$uibModalInstance', 'DebtorGroupService', 'NotifyService',
+  'CashboxService', 'SessionService', 'data',
+  'AccountStoreService', 'DebtorService', 'PatientInvoiceService'
+];
+
+// Import transaction rows for a convention payment
+function ConventionPaymentKitController(Instance, DebtorGroup, Notify, Cashbox, Session, Data, AccountStore, Debtors, Invoices) {
+  var vm = this;
+
+  // global variables
+  vm.enterprise = Session.enterprise;
+  vm.gridOptions = {};
+  vm.tool = Data;
+
+  // expose to the view
+  vm.selectGroupInvoices = selectGroupInvoices;
+  vm.close = Instance.close;
+  vm.import = submit;
+
+  // accounts from store
+  AccountStore.accounts().then(function (data) {
+    vm.accounts = data;
+  })
+  .catch(Notify.handleError);
+
+  // debtors from store
+   Debtors.store()
+   .then(function (data) {
+     vm.debtors = data;
+   })
+   .catch(Notify.handleError);
+
+  //  optimization with `Store` will be well
+   Invoices.read()
+   .then(function (data) {
+     vm.invoices = data;
+   })
+   .catch(Notify.handleError);
+
+  // load debtors
+  Debtors.read()
+  .then(function (list) {
+    vm.debtorList = list;
+  })
+  .catch(Notify.handleError);
+
+  // load conventions
+  DebtorGroup.read(null, { is_convention: 1})
+  .then(function (list) {
+    vm.conventionGroupList = list;
+  })
+  .catch(Notify.handleError);
+
+  // load cashboxes
+  Cashbox.read(null, { detailed: 1, is_auxiliary: 0})
+  .then(function (list) {
+    vm.cashboxList = list;
+  })
+  .catch(Notify.handleError);
+
+  // get debtor group invoices
+  function selectGroupInvoices(convention) {
+    DebtorGroup.invoices(convention.uuid, { is_convention: 1})
+    .then(function (list) {
+      // invoices
+      vm.gridOptions.data = list || [];
+
+      // total amount
+      vm.totalInvoices = vm.gridOptions.data.reduce(function (current, previous) {
+        return current + previous.balance;
+      }, 0);
+    })
+    .catch(Notify.handleError);
+  }
+
+  // generate transaction rows
+  function generateTransactionRows(result) {
+    vm.rows = [];
+
+    var cashboxAccount = vm.accounts.get(result.cashbox.account_id);
+    var conventionAccount = vm.accounts.get(result.convention.account_id);
+    var entityAccount;
+    var invoiceReference;
+    var index;
+
+    // cashbox row
+    vm.rows.push(generateRow());
+    vm.rows[0].account = cashboxAccount;
+    vm.rows[0].debit = vm.totalSelected;
+    vm.rows[0].credit = 0;
+
+    // cashbox and invoices rows
+    for (var i = 0; i < result.invoices.length; i++) {
```

Comment:
Did you know that `Array.prototype.forEach` has three parameters?  [Documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach).

So, your code can be much more semantic:
```js
result.invoices.forEach(function (invoice, index) {
  entityAccount = formatEntity(invoice);
  invoiceReference = getInvoiceReference(invoice);
});
```

You actually don't need `index` or `i`!  `Array.prototype.push()` returns the next index.  Like this:

```js
var index = vm.rows.push(generateRow());
// index is now the length of rows.
```